src/main/java/gui/Analysis.java

```diff
@@ -0,0 +1,140 @@
+            cols.add("All");
+
+            List<String> dataSetCols = dataSet.getColumnNames();
+
+            for(int name=0; name<dataSetCols.size(); ++name){
+                cols.add(dataSetCols.get(name));
+            }
+
+            DataSetViewInfoHolder holder = new DataSetViewInfoHolder();
+            holder.name = dataSet.getName();
+            holder.names = cols;
+
+            dataSets.add(holder);
+        }
+
+        model.addAttribute("dataSets", dataSets);
+        return "analysis_index";
+    }
+
+    @PostMapping
+    public String handleAnalysisForm(@Valid AnalysisFormWrapper formWrapper, Errors errors){
+
+        // validate form
+        if (errors.hasErrors()) {
+            System.out.println("Form has errors...");
+            return "/analysis";
+        }
+
+        // Try to figure out what sort of
+        // analysis we want to do
+        if(formWrapper.eda != null){
+
+            this.computeDataSetStatistics(formWrapper);
+        }
+        else if(formWrapper.linear_regression != null){
+
+        }
+        else if(formWrapper.non_linear_regression != null){
+
+        }
+        else if(formWrapper.logistic_regression != null){
+
+        }
+        else if(formWrapper.kmeans_clustering != null){
+
+        }
+        else{
+
+            // return an error message that no valid action
+            // was selected
+        }
+
+        // redirect to the analysis page again
+        return "redirect:/analysis-result?taskName=" + "Mean";
+    }
+
+    protected void computeDataSetStatistics(AnalysisFormWrapper formWrapper){
+
+
+        System.out.println("================");
+        System.out.println("computeDataSetStatistics...");
+        System.out.println("================");
+        System.out.println("Filename exploreDataSet: "+formWrapper.eda);
+        System.out.println("Filename exploreDataSet: "+formWrapper.dataSetName);
+        System.out.println("Column Name: " + formWrapper.colName);
+
+        // get the name of the dataset
+        String dataSetName = formWrapper.dataSetName;
+        IDataSet dataSet = JStatGuiGlobalData.dataSetContainer.getDataSet(dataSetName);
+
+        if(formWrapper.colName == "All"){
+
+            List<String> dataSetCols = dataSet.getColumnNames();
+            String[] names = new String[dataSetCols.size()];
+
+            for(int i=0; i<names.length; ++i){
+                names[i] = dataSetCols.get(i);
+            }
+
+            // submit it to the pool
+            TaskBase task = new ComputeDescriptiveStatisticsTask("Mean", dataSet, names);
+            JStatGuiGlobalData.workersPool.submit(task);
+            JStatGuiGlobalData.tasks.add(task);
+
+        }
+        else{
+
+            // submit it to the pool
+            TaskBase task = new ComputeDescriptiveStatisticsTask("Mean", dataSet, formWrapper.colName);
+            JStatGuiGlobalData.workersPool.submit(task);
+            JStatGuiGlobalData.tasks.add(task);
+        }
+    }
+}
```

Comment:
as I told you we need to revisit the controllers as far as the values are concerned.

Reply:
Yes i agree but to my mind the controllers are just to serve requests this is not a request you just pust a computation specified by the ```task```. In fact this is not work pool is just an extra array to query in O(N) time but still need to figure out how are we going to clean this from the tasks that have been executed :) :)

Reply:
In any case I think we need an ```Analysis``` controller to start any kind of analysis and then redirect to the analogous result page as in  ```"redirect:/analysis-result?taskName="+taskName;```